src/main/java/com/lelek/dbAccess/service/SenderService.java

```diff
@@ -0,0 +1,83 @@
+package com.lelek.dbAccess.service;
+
+import com.lelek.dbAccess.dao.MessageDao;
+import com.lelek.dbAccess.dto.MessageDto;
+import com.lelek.dbAccess.model.MySimpleMailMessage;
+import lombok.AllArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Setter
+@Service
+@AllArgsConstructor
+public class SenderService extends Thread {
+
+    private static boolean stop = false;
+    private static SimpleDateFormat format = new SimpleDateFormat("dd.MM.yyyy HH:mm");
+
+    private JavaMailSender javaMailSender;
+
+    private MessageDao messageDao;
+
+    private MySimpleMailMessage mapToSimpleMailMessage(MessageDto messageDto) {
+        MySimpleMailMessage mySimpleMailMessage = new MySimpleMailMessage();
+        mySimpleMailMessage.setId(messageDto.getId());
+        mySimpleMailMessage.setTo(messageDto.getTo());
+        mySimpleMailMessage.setCc(messageDto.getTo());
+        mySimpleMailMessage.setBcc(messageDto.getTo());
+        mySimpleMailMessage.setSubject(messageDto.getSubject());
+        mySimpleMailMessage.setText(messageDto.getText());
+        try {
+            mySimpleMailMessage.setSentDate(format.parse(messageDto.getDate()));
+        } catch (ParseException e) {
+            log.error("error " + e);
+        }
+        mySimpleMailMessage.setSent(messageDto.isSent());
+        return mySimpleMailMessage;
+    }
+
+    private MessageDto mapToDto(MySimpleMailMessage mySimpleMailMessage) {
+        return new MessageDto(mySimpleMailMessage.getId(),
+                mySimpleMailMessage.isSent(),
+                Objects.requireNonNull(mySimpleMailMessage.getTo())[0],
+                mySimpleMailMessage.getSubject(),
+                mySimpleMailMessage.getText(),
+                format.format(mySimpleMailMessage.getSentDate()));
+    }
+
+    private void send() throws IOException {
+        List<MySimpleMailMessage> messagesToSend = messageDao.getMessages().stream()
+                .map(this::mapToSimpleMailMessage)
+                .filter(message -> !message.isSent())
+                .filter(message -> Objects.requireNonNull(message.getSentDate()).getTime() <= System.currentTimeMillis())
```

Comment:
If you need non null value make this validation with one more filter above.
Filter condition of stream should be as simple as possible.